File: cpp/57b_1629A_DownloadMoreRAM.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    int t, n, k, a, b;
    vector<int> ans;
    
    cin>>t;

    for (int i = 0; i < t; i++){
        cin>>n>>k;
        vector<vector<int>> arr;
        for (int j = 0; j < n; j++){
            cin>>a;
            arr.push_back({a});
        }
        for(int j = 0; j < n; j++){
            cin>>b;
            arr[j].push_back(b);
        }
        sort(arr.begin(), arr.end(), [](vector<int> x, vector<int> y){
            return x[0] < y[0];
        });
        for(int j = 0; j < n; j++){
            if(arr[j][0] > k){
                continue;
            }
            else{
                k += arr[j][1];
            }
        }
        ans.push_back(k);
    }
    for(auto i : ans){
        cout<<i<<"\n";
    }
    return 0;
}

File: python/57b_1629A_DownloadMoreRAM.py
def solve(a, b, k, n):
    arr = []
    for i in range(n):
        a[i] = int(a[i])
        b[i] = int(b[i])
        arr.append([a[i],b[i]])
    arr.sort()
    for i in range(n):
        if arr[i][0] > k:
            continue
        else:
            k += arr[i][1]
    return k

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n, k = input().split()
        n, k = int(n), int(k)
        a = input().split()
        b = input().split()
        ans.append(solve(a, b, k, n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()